entry_types/scrolled/package/src/frontend/Entry.js

```diff
@@ -17,12 +18,20 @@ export default withInlineEditingDecorator('EntryDecorator', function Entry(props
   const entryStructure = useEntryStructure();
 
   const setCurrentSectionIndex = useCallback(index => {
+    const getSectionChapterTitle = (sectionId, delimiter = '') => entryStructure.filter(
+      chapter => chapter.sections.some(section => section.id === sectionId)
+    )[0]?.title + delimiter;
+
     if (window.parent !== window) {
       window.parent.postMessage({type: 'CHANGE_SECTION', payload: {index}}, window.location.origin);
     }
 
+    events.trigger('page:change', {configuration: {
+      title: getSectionChapterTitle(index, ', ') + 'Section ' + index
+    }, index: index});
+
     setCurrentSectionIndexState(index);
-  }, [setCurrentSectionIndexState]);
+  }, [entryStructure, setCurrentSectionIndexState]);
```

Comment:
We can test that the desired events with the desired data are being triggered using a similar test like [the one for the `CHANGE_SECTION` message](https://github.com/codevise/pageflow/blob/5a650782dff8f58252ea084e0bd55d8eafca49b5/entry_types/scrolled/package/spec/frontend/features/changeSectionMessage-spec.js#L4).

Awais also brought to my attention that the callback might be invoked multiple times in certain scenarios. To prevent triggering multiple events, we could use the [technique described in the hooks FAQ](https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state) to track the previous section index and only trigger events when the index changes. We already have a [`usePrevious` hook in our code base](https://github.com/codevise/pageflow/blob/5a650782dff8f58252ea084e0bd55d8eafca49b5/entry_types/scrolled/package/src/frontend/usePrevious.js#L3) that we can use here. Let's also extract this logic into a separate file like `useSectionChangeEventEmitter` etc.